crates/optimism/primitives/src/proof.rs

```diff
@@ -0,0 +1,330 @@
+use reth_primitives::{
```

Comment:
we need module-level docs here

Reply:
Didn't understood what you are suggesting here what do you mean by module level docs ?